test/pipeline_test_runner.cc

```diff
@@ -267,11 +268,21 @@ vector<ast::ParsedFile> index(core::GlobalState &gs, absl::Span<core::FileRef> f
 
         // Desugarer
         ast::ParsedFile desugared;
-        {
-            core::UnfreezeNameTable nameTableAccess(gs); // enters original strings
+        switch (parser) {
+            case realmain::options::Parser::ORIGINAL: {
+                core::UnfreezeNameTable nameTableAccess(gs); // enters original strings
+
+                core::MutableContext ctx(gs, core::Symbols::root(), file);
+                desugared = testSerialize(gs, ast::ParsedFile{ast::desugar::node2Tree(ctx, move(nodes)), file});
+                break;
+            }
+            case realmain::options::Parser::PRISM: {
+                core::UnfreezeNameTable nameTableAccess(gs); // enters original strings
 
-            core::MutableContext ctx(gs, core::Symbols::root(), file);
-            desugared = testSerialize(gs, ast::ParsedFile{ast::desugar::node2Tree(ctx, move(nodes)), file});
+                core::MutableContext ctx(gs, core::Symbols::root(), file);
+                desugared = testSerialize(gs, ast::ParsedFile{ast::prismDesugar::node2Tree(ctx, move(nodes)), file});
+                break;
+            }
```

Comment:
I'm not sure why this doesn't call `runDesugar`, and instead calls `node2Tree` directly. Perhaps we could address that in a follow-up.